Training/Training/TestCase.c

```diff
@@ -0,0 +1,158 @@
+   char* output1[] = {
+      "00000000000000000000000000000000",
+      "00000000000000000000001010001011",
+      "11111111111111111111110111011111",
+      "00000000000000000000001111100111",
+      "11111111111111111110110011110101",
+      "00000000000000000011110110001100",
+      "11111111111111111110110111010010",
+      "11111111111111111111111111111111",
+      "10000000000000000000000000000001" };
+   TestDecToBin (input1, output1, length1);
+   char* output2[] = {
+        "00000000",
+        "0000028B",
+        "FFFFFDDF",
+        "000003E7",
+        "FFFFECF5",
+        "00003D8C",
+        "FFFFEDD2",
+        "FFFFFFFF",
+        "80000001" };
+   TestDecToHex (input1, output2, length1);
+   printf ("\n");
+//test cases for Hexadecimal conversion
+   int input2[] = { 1,255,1023,16384,2147483647,-1,-5,-2147483647,-2147483648ll };
+   int length2 = sizeof (input2) / sizeof (input2[0]);
+   char* output3[] = {
+      "00000000000000000000000000000001",
+      "00000000000000000000000011111111",
+      "00000000000000000000001111111111",
+      "00000000000000000100000000000000",
+      "01111111111111111111111111111111",
+      "11111111111111111111111111111111",
+      "11111111111111111111111111111011",
+      "10000000000000000000000000000001",
+      "10000000000000000000000000000000" };
+   TestDecToBin (input2, output3, length2);
+   char* output4[] = {
+      "00000001",
+      "000000FF",
+      "000003FF",
+      "00004000",
+      "7FFFFFFF",
+      "FFFFFFFF",
+      "FFFFFFFB",
+      "80000001",
+      "80000000" };
+   TestDecToHex (input2, output4, length2);
+//User input validation
+   while (1) {
+      long long int dec = 0;
+      dec = GetIntegerOnly ();
+      if (dec != INT_MIN) {
+         char* binNum = DecToBin ((unsigned int)dec);
+         printf ("Binary value(0b): "ANSI_COLOR_BLUE"%s"ANSI_RESET_ALL, binNum);
+         char* hexNum = DecToHex ((unsigned int)dec);
+         printf ("\nHexadecimal value(0x): "ANSI_COLOR_MAGENTA"%s\n"ANSI_RESET_ALL, hexNum);
+      }
+   }
+}
+
+void TestDecToBin (int input[], char* outputBin[], int arrLength) {
+   printf ("-----Decimal to Binary-----\n");
+   for (int i = 0; i < arrLength; i++) {
+      printf ("Test Case%d: Input--> %-11d ", i + 1, input[i]);
+      char* binResult1 = DecToBin (input[i]);
+      printf ((strcmp (binResult1, outputBin[i]) == 0) ? "%11s", ANSI_COLOR_GREEN"PASS\n"ANSI_RESET_ALL : "%11s", ANSI_COLOR_RED"FAIL\n"ANSI_RESET_ALL);
+   }
+}
+
+void TestDecToHex (int input[], char* outputHex[], int arrLength) {
+   printf ("\n-----Decimal to Hexadecimal-----\n");
+   for (int i = 0; i < arrLength; i++) {
+      printf ("Test Case%d: Input--> %-11d ", i + 1, input[i]);
+      char* binResult2 = DecToHex (input[i]);
+      printf ((strcmp (binResult2, outputHex[i]) == 0) ? "%11s", ANSI_COLOR_GREEN"PASS\n"ANSI_RESET_ALL : "%11s", ANSI_COLOR_RED"FAIL\n"ANSI_RESET_ALL);
+   }
+}
+
+long long int GetIntegerOnly () {

```

Comment:
You don't need to return long long int, if it goes beyond a certain length, you can print a "input beyond range error"